pkg/version/version.go

```diff
@@ -41,12 +44,38 @@ func (info Info) String() string {
 // what code a binary was built from.
 func Get() Info {
 	return Info{
-		GitVersion:   gitVersion,
-		GitCommit:    gitCommit,
-		GitTreeState: gitTreeState,
-		BuildDate:    buildDate,
-		GoVersion:    runtime.Version(),
-		Compiler:     runtime.Compiler,
-		Platform:     fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+		GitVersion:            gitVersion,
+		GitAbbreviativeCommit: gitAbbreviativeCommit,
+		GitCommit:             gitCommit,
+		GitTreeState:          gitTreeState,
+		BuildDate:             buildDate,
+		GoVersion:             runtime.Version(),
+		Compiler:              runtime.Compiler,
+		Platform:              fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
 	}
 }
+
+// NewCollector returns a collector that exports metrics about current version
+// information.
+func NewCollector(program string) prometheus.Collector {
+	return prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: program,
```

Comment:
Maybe we can omit the component name?
Because the metric emitted from a component indicates it's build info.

Reply:
The metrcs specification defines different metrics names, and is not distinguished by different tags of the same metrics name.
ref: https://github.com/prometheus/client_golang/blob/main/prometheus/collectors/version/version.go#L28-L30